utils/perl5lib/Config/ConfigCompsetGrid.pm

```diff
@@ -112,12 +112,12 @@ sub getCompsetLongname
 	    $file =~ s/\$MODEL/$model/;
 	    $outstr .= "$file\n";
 	}
-	$logger->logdie ($outstr);
+	$logger->logwarn ($outstr);
```

Comment:
Changing this from an error to a warning does not feel like a robust fix to me. I see two problems with this:

(1) If this is called for a case that uses -compset (rather than -user_compset), then this should generate an error, not a warning. Otherwise you end up with a cryptic death if you try to create a case with a non-existent compset:

```
Use of uninitialized value $pes_setby in concatenation (.) or string at
	./create_newcase line 313 (#1)
    (W uninitialized) An undefined value was used as if it were already
    defined.  It was interpreted as a "" or a 0, but maybe it was a mistake.
    To suppress this warning assign a defined value to your variables.
    
    To help you figure out what was undefined, perl will try to tell you the
    name of the variable (if any) that was undefined. In some cases it cannot
    do this, so it also tells you what operation you used the undefined value
    in.  Note, however, that perl optimizes your program and the operation
    displayed in the warning may not necessarily appear literally in your
    program.  For example, "that $foo" is usually optimized into "that "
    . $foo, and the warning will refer to the concatenation (.) operator,
    even though there is no . in your program.
    
Use of uninitialized value $components in split at ./create_newcase line 328 (#1)
grid longname is : a%0.9x1.25_l%0.9x1.25_oi%gx1v6_r%r05_m%gx1v6_g%null_w%null
Uncaught exception from user code:
	ERROR create_newcase: config_file unset does not exist at ./create_newcase line 415.
 at ./create_newcase line 415
```

(2) If this is called when using -user_compset, then it should not even generate a warning - because the lack of existence of a compset match will be the usual, expected case. Many people wonder if they have done something wrong when they see a warning message, so we should not generate warning messages for the expected case.

Would it be possible to fix both of these problems by giving a new argument to getCompsetLongname, like "require_match"? require_match would be true if calling this with a case where -compset was provided, false for a case where -user_compset was provided. If require_match is true, then this generates an ERROR message. If false, this doesn't generate any message.